mkosi/gentoo.py

```diff
@@ -0,0 +1,508 @@
+        if not os.path.isfile(stage3_tar_path):
+            MkosiPrinter.print_step(f"Fetching {stage3_url_path}")
+            os.makedirs(os.path.dirname(stage3_tar_path), exist_ok=True)
+            urllib.request.urlretrieve(stage3_url_path, stage3_tar_path)
+
+        with tarfile.open(stage3_tar_path) as tfd:
+            MkosiPrinter.print_step(f"Extracting {os.path.basename(stage3_tar)}")
+            # GENTOO: usrmerge mess
+            bins = []
+            usr = []
+            usrreg = "^[.][/](etc|usr|var)"
+            binreg = "^[.][/](sbin|bin|lib|lib64)"
+            devreg = "^[.][/]dev"
+            runreg = "^[.][/]run"
+
+            members = tfd.getmembers()
+            usrpath = root / "usr"
+
+            self.UNINSTALL_IGNORE = {
+                "/bin",
+                "/sbin",
+                "/lib",
+                "/lib64",
+            }
+
+            ignore = ["./var/run", "./var/lock", "./bin/awk"]
+            for dir in self.UNINSTALL_IGNORE:
+                ignore += [dir] + [f"./usr/{dir}"]
+
+            for tarinfo in members:
+                if tarinfo.name in ignore or re.match(devreg, tarinfo.name) or re.match(runreg, tarinfo.name):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> /dev/null")
+                    continue
+                elif (
+                    tarinfo.issym()
+                    and (os.path.dirname(tarinfo.name) == "./usr/bin")
+                    and (os.path.dirname(tarinfo.linkname) == "../../bin")
+                ):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> {tarinfo.linkname} -> /dev/null")
+                    continue
+                elif re.match(usrreg, tarinfo.name):
+                    usr.append(tarinfo)
+                elif re.match(binreg, tarinfo.name):
+                    bins.append(tarinfo)
+                else:
+                    print(f"{tarinfo.name} -> /dev/null")
+                    continue
+
+            try:
+                tfd.extractall(usrpath, bins, numeric_owner=True)
+                tfd.extractall(root, usr, numeric_owner=True)
+            except FileExistsError:
+                pass
+
+    def set_profile(self, args: CommandLineArguments, root: str) -> None:
+        # be careful not to overwrite user's profile, (via skeleton)
+        profile_path = root / PROFILE_PATH
+        if not os.path.islink(profile_path):
+            MkosiPrinter.print_step(f"{args.distribution} setting Profile")
+            os.makedirs(root / USER_CONFIG_PATH, exist_ok=True)
+            os.symlink(self.portage_cfg["PORTDIR"] / self.arch_profile, profile_path)
+
+    def set_default_repo(self, root: str) -> None:
+        os.makedirs(root / USER_CONFIG_PATH / "repos.conf", exist_ok=True)
+        with open(root / USER_CONFIG_PATH / "repos.conf" / "eselect-repo.conf", "w") as f:
+            f.write(
```

Comment:
```py
eselect_repo_conf = root / USER_CONFIG_PATH / "repos.conf/eselect-repo.conf"
eselect_repo_conf.parent.mkdir(exist_ok=True)
eselect_repo_conf.write_text(dedent(
       f"""\
       ...
       """))
```
or so.

Reply:
changed as many as my eyes cough; please let me know if you spot any.